test/test_addmult.c

```diff
@@ -313,14 +315,23 @@ void test_arrlss(void **state) {
     setcontest("arrl_ss");
 
     setup_multis("SC\nSCV\n");
-    strcpy(ssexchange, "SCV");
-    set_this_qso("");   // NOTE: ssexchange is not part of qso_t
+    strcpy(mult1_value, "S");   // incomplete value (normally doesn't happen)
+    set_this_qso("");   // NOTE: mult1_value is not part of qso_t
     addmult(this_qso);
-    strcpy(ssexchange, "97A23SCV");
+    assert_int_equal(nr_multis, 0);
+#if 0
```

Comment:
@dl1jbe Tom, could you check this? Here an invalid mult ("SCX") is provided but it counts as "SC". (the test fails, hence the commeting out) This is related to the looking for the longest `get_matching_length()` in `addmult`. Is this really correct?

Reply:
Will do, but give me a day

Reply:
Original code from Rein picked up first matching multi (even in an exchange without spaces) - so if SC and SCX are in that order in the list of possible mults it choose SC even SCX where in the exchange. It got changed around '18 to pick up the longest matching multi. But it seems that having a wrong multi in the exchange got not handled properly.

In your actual code you have the section as standalone string in 'mult1_value'. With that we should convert the handling in 'addmult()' from finding the longest match to a check for a full match ('get_exact_mult_index()'). That would be better and fixes your problem in the test code.

Reply:
Pushed fix. Probably `get_matching_length()` could be dropped later on.